bpf/bpf_host.c

```diff
@@ -239,13 +239,13 @@ handle_ipv6(struct __ctx_buff *ctx, __u32 secctx, const bool from_host)
 	}
 #endif /* ENABLE_HOST_FIREWALL */
 
+skip_host_firewall:
 #if defined(NO_REDIRECT) && !defined(ENABLE_HOST_ROUTING)
 	/* See IPv4 case for NO_REDIRECT/ENABLE_HOST_ROUTING comments */
 	if (!from_host)
 		return CTX_ACT_OK;
 #endif /* NO_REDIRECT && !ENABLE_HOST_ROUTING */
 
-skip_host_firewall:
```

Comment:
It strikes me as a fairly exotic bug, but sure why not.

Reply:
Well, what's the exact impact? I haven't checked and there was no discussion of that in the commit description :wink:

Reply:
> Well, what's the exact impact?

Just from looking at the code? Your favorite - asymmetry ;). We can miss cases (`NO_REDIRECT && !ENABLE_HOST_ROUTING`) where a packet **needs** to go from-netdev-to-pod via the stack, as (1) iptables is active and (2) the from-pod-to-netdev path will pass replies via the stack (due to `!ENABLE_HOST_ROUTING`).

For which of those allow-listed ICMPv6 messages that is actually relevant? :shrug:

Reply:
Ok. Packet asymmetry is usually only an issue for service handling and I'm guessing that won't be an issue with ICMPv6 packets. So probably okay given no user impact?

Reply:
> So probably okay given no user impact?

That was my gut feeling, yes. Will leave it as first patch in the split-up series then, so that we can easily cherry-pick it later if necessary.